### ISSUE TITLE
gl_SampleMask specified as unsized in some cases, breaking ANGLE

### ISSUE THREAD

**Issue by null77** [KhronosGroup/glslang#2441]
We're running into a bug with ANGLE's implementation of `OES_sample_variables`. See spec here:

https://www.khronos.org/registry/OpenGL/extensions/OES/OES_sample_variables.txt

The spec (and GLES 3.2) define `gl_SampleMaskIn` as 

```
        in  highp    int  gl_SampleMaskIn[(gl_MaxSamples+31)/32];
```

Desktop GLSL define it has:

```
in int gl_SampleMaskIn[];
```

This poses a huge problem in ANGLE because we need to access `gl_SampleMaskIn` dynamically for ES 3.2 / sample variables correctness and it's currently impossible for us to switch to GLSL ES input for gslalng.

The specific error case is:

```
   Except for the last declared member of a shader storage block (section 4.3.9 “Interface Blocks”),
    the size of an array must be declared (explicitly sized) before it is indexed with anything other than an integral constant expression.
    The size of any array must be declared before passing it as an argument to a function.
    Violation of any of these rules result in compile-time errors.
    It is legal to declare an array without a size (unsized) and then later redeclare the same name as an array of the same type and specify a size,
    or index it only with integral constant expressions (implicitly sized).
```

Shader snippet:

```
    void main(){
        for (int _ui = 0; (_ui < 1); (++_ui))
        {
            (gl_SampleMask[_ui] = -1431655766);
        }
        float _ublue = float(gl_SampleID);
        (_ufragColor = vec4(0.0, 1.0, _ublue, 1.0));
}
```

Both glslang and SPIR-V support this functionality. How can we get past this validation?

**Comment by johnkslang** [comment]
### Answer 1:

Generally, unsized arrays get sized by the time glslang is done linking everything, and are not actually run-time arrays.  Maybe there can a configuration constant for how large this needs to be (see content of `.conf` file, or `TResources`) and glslang sets it to that size if nothing else sized it.

### Answer 2:

How big can this actually be?  E.g., 2?

I think it's a very small number, and unrolling the loop once or twice might solve the problem in practice.

**Comment by arcady-lunarg** [comment]
This was fixed by #3399.
